core/config/config.go

```diff
@@ -0,0 +1,164 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/pelletier/go-toml"
+	"github.com/tidwall/sjson"
+	"gopkg.in/yaml.v2"
+)
+
+type Handler struct {
+	cfg      interface{}
+	filePath string
+	fileExt  string
+}
+
+func New(path string, cfg interface{}) *Handler {
+	// TODO: Add code to check if cfg is of type: struct
+
+	// fmt.Println(reflect.TypeOf(cfg))
+	// t := reflect.ValueOf(cfg).Kind()
+	// if t != reflect.Struct {
+	// 	return fmt.Errorf("The object passed in as argument is not a struct.", t)
+	// }
+
+	h := &Handler{
+		filePath: path,
+		cfg:      cfg,
+		fileExt:  filepath.Ext(path),
+	}
+
+	return h
+}
+
+func (h *Handler) Load() error {
+	// Add code to read local config file
+
+	buf, err := readFile(h.filePath)
+	if err != nil {
+		return err
+	}
+
+	// Add code to store values (read from file) to the host config struct [ie. Unmarshal]
+
+	err = h.unmarshal(buf, h.cfg)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Handler) Save() error {
```

Comment:
In it's current form (maybe you have plans to change it) but this Save function overwrites all data in the config file right? Wouldn't we just want to save the values defined in the host's config struct and leave the plugin data.